### DISCUSSION TITLE
Is depth sorting needed for transparency?

### DISCUSSION THREAD

**Discussion by jwwalker** [google/filament#3369] [Q&A] [answered]
I read the section of "Physically Based Rendering in Filament" on transparency and translucency, but I could use more information on how to handle it as a practical matter.  If i want to render a number of geometry parts, some of which are translucent surfaces, do I need to worry about handing them to Filament in back to front order, or does Filament do some magic to make that unnecessary?

**Comment by romainguy** [comment, marked as answer]
Filament will sort renderables back to front, make sure you provide the proper bounding boxes for this to work well though. Note that overlapping transparent objects won't necessarily render in the order you expect. You can play with priorities or use the advanced transparency modes we provide like twoPassesTwoSides for instance: https://google.github.io/filament/Materials.html#materialdefinitions/materialblock/blendingandtransparency:transparency

**Reply by jwwalker** [reply]
What if I had a single triangular mesh representing something like a drinking glass, such that when viewed from a typical angle you would be looking through both the front and the back surface?  Would that need to be broken up into smaller pieces with smaller bounding boxes?

**Reply by romainguy** [reply]
See my link above
